### PR TITLE
Use concat to define the process plugin config.

### PR DESCRIPTION
**Pull request by bzed**
Adding a parser function to keep it compatible with the old way
to define process matches.

### CODE REVIEW

**Comment by bzed** [comment]
Somebody please fix the spec tests, I'm blind and don't spot the error.

File: manifests/plugin/processes.pp [hunk 1]
```diff
@@ -10,7 +11,43 @@
 
   collectd::plugin {'processes':
     ensure   => $ensure,
-    content  => template('collectd/plugin/processes.conf.erb'),
+    order    => $order,
     interval => $interval,
   }
+
+  concat{"${collectd::params::plugin_conf_dir}/processes-config.conf":
```
Comment (txaj) [inline manifests/plugin/processes.pp]: I think you're missing a ```include collectd::params```

Comment (bzed) [inline manifests/plugin/processes.pp]: Good catch!

**Comment by blkperl** [comment]
I'm slightly confused on what this patch does? Do the docs need to be updated as well?

**Comment by bzed** [comment]
I've updated the documentaion, hope its clear now?

**Comment by blkperl** [comment]
Ok it makes sense now. Thanks for the contribution

**Pull request merged**